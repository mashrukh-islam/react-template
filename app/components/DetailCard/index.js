/**
 *
 * DetailCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card, Statistic } from 'antd';

import { T } from '@components/T';

const Container = styled.div`
  && {
    display: flex;
    flex-direction: column;
    max-width: ${props => props.maxWidth}em;
    width: 100%;
    margin: 0 auto;
  }
`;

const Row = styled.div`
  && {
    display: flex;
  }
`;
const Col = styled.div`
  && {
    flex: ${props => props.size};
  }
`;
function DetailCard({ maxWidth, trackDetails }) {
  const renderTrackTime = millis => {
    var minutes = Math.floor(millis / 60000);
    var seconds = ((millis % 60000) / 1000).toFixed(0);
    return minutes + ':' + (seconds < 10 ? '0' : '') + seconds;
  };

  return (
    <Container maxWidth={maxWidth} data-testid="detail-card">
      <Card
        data-testid="artwork-card"
        cover={<img src={trackDetails.artworkUrl100} />}
        actions={[
          <a href={trackDetails.artistViewUrl} key="artist" data-testid="artist-view-link">
            <T id="view_artist" data-testid="view-artist-text" />
          </a>,
          <a href={trackDetails.collectionViewUrl} key="artist" data-testid="album-view-link">
            <T id="view_album" data-testid="view-album-text" />
          </a>,
          <a href={trackDetails.trackViewUrl} key="artist" data-testid="track-view-link">
            <T id="view_track" data-testid="view-track-text" />
          </a>
        ]}
      >
        <Row>
          <Col size={1}>
            <Statistic title={<T id="track_name" />} value={trackDetails.trackCensoredName} />
            <Statistic title={<T id="album" />} value={trackDetails.collectionName} />
          </Col>
          <Col size={1}>
            <Statistic title={<T id="track_time" />} value={renderTrackTime(trackDetails.trackTimeMillis)} />
            <Statistic title={<T id="genre" />} value={trackDetails.primaryGenreName} />
          </Col>
        </Row>
        <Row>
          <Col size={1}>
            <Statistic title={<T id="explicitness" />} value={trackDetails.trackExplicitness} />
          </Col>
          <Col size={1}>
            <Statistic title={<T id="artist" />} value={trackDetails.artistName} />
          </Col>
        </Row>
      </Card>
    </Container>
  );
}

DetailCard.propTypes = {
  maxWidth: PropTypes.number,
  trackDetails: PropTypes.object
};

DetailCard.defaultProps = {
  maxWidth: 50,
  trackDetails: null
};

export default DetailCard;
