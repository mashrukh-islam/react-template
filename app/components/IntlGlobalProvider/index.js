// eslint-disable-next-line
import React from 'react';
import { useIntl } from 'react-intl';

// 'intl' service singleton reference
let intl;

export function IntlGlobalProvider({ children }) {
  intl = useIntl(); // Keep the 'intl' service reference
  return children;
}

// Getter function to expose the read-only 'intl' service
export function appIntl() {
  return intl;
}
// Set intl value for translate in saga tests
export const setIntl = i => {
  intl = i;
};

export const translate = (id, values = {}) => intl.formatMessage({ id }, values);
