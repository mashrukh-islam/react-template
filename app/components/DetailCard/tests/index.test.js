/**
 *
 * Tests for DetailCard
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import DetailCard from '../index';
import { translate } from '@components/IntlGlobalProvider';

describe('<DetailCard />', () => {
  let trackDetails;

  beforeEach(() => {
    trackDetails = {
      artistViewUrl: 'https://music.apple.com/us/artist/opeth/3196120?uo=4',
      artworkUrl100: 'https://music.apple.com/us/artist/opeth/3196120?uo=4',
      collectionViewUrl: 'https://music.apple.com/us/artist/opeth/3196120?uo=4',
      trackViewUrl: 'https://music.apple.com/us/artist/opeth/3196120?uo=4'
    };
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<DetailCard trackDetails={trackDetails} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 DetailCard component', () => {
    const { getAllByTestId } = renderWithIntl(<DetailCard trackDetails={trackDetails} />);
    expect(getAllByTestId('detail-card').length).toBe(1);
  });

  it('should ensure that actions are rendered', () => {
    const { getByTestId } = renderWithIntl(<DetailCard trackDetails={trackDetails} />);
    expect(getByTestId('view-album-text').textContent).toEqual(translate('view_album'));
    expect(getByTestId('view-artist-text').textContent).toEqual(translate('view_artist'));
    expect(getByTestId('view-track-text').textContent).toEqual(translate('view_track'));
  });

  it('should ensure that links have correct source', () => {
    const { getByTestId } = renderWithIntl(<DetailCard trackDetails={trackDetails} />);
    expect(getByTestId('artist-view-link').getAttribute('href')).toBe(trackDetails.artistViewUrl);
    expect(getByTestId('album-view-link').getAttribute('href')).toBe(trackDetails.collectionViewUrl);
    expect(getByTestId('track-view-link').getAttribute('href')).toBe(trackDetails.trackViewUrl);
  });

  it('should ensure that artwork is rendered', () => {
    const { getAllByTestId } = renderWithIntl(<DetailCard trackDetails={trackDetails} />);
    expect(getAllByTestId('artwork-card')).toHaveLength(1);
  });
});
