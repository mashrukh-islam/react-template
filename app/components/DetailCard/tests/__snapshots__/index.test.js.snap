// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="DetailCard__Container-sc-1e9kpob-0 olfZV"
      data-testid="detail-card"
    >
      <div
        class="ant-card ant-card-bordered"
        data-testid="artwork-card"
      >
        <div
          class="ant-card-cover"
        >
          <img
            src="https://music.apple.com/us/artist/opeth/3196120?uo=4"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="DetailCard__Row-sc-1e9kpob-1 gRwRoj"
          >
            <div
              class="DetailCard__Col-sc-1e9kpob-2 dngWDh"
              size="1"
            >
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Track Name
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Album
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div
              class="DetailCard__Col-sc-1e9kpob-2 dngWDh"
              size="1"
            >
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Track Time
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    NaN:NaN
                  </span>
                </div>
              </div>
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Genre
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="DetailCard__Row-sc-1e9kpob-1 gRwRoj"
          >
            <div
              class="DetailCard__Col-sc-1e9kpob-2 dngWDh"
              size="1"
            >
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Explicitness
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div
              class="DetailCard__Col-sc-1e9kpob-2 dngWDh"
              size="1"
            >
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Artist
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <ul
          class="ant-card-actions"
        >
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <a
                data-testid="artist-view-link"
                href="https://music.apple.com/us/artist/opeth/3196120?uo=4"
              >
                <p
                  class="T__StyledText-gjlic1-0 cbAvsY"
                  data-testid="view-artist-text"
                >
                  View Artist
                </p>
              </a>
            </span>
          </li>
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <a
                data-testid="album-view-link"
                href="https://music.apple.com/us/artist/opeth/3196120?uo=4"
              >
                <p
                  class="T__StyledText-gjlic1-0 cbAvsY"
                  data-testid="view-album-text"
                >
                  View Album
                </p>
              </a>
            </span>
          </li>
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <a
                data-testid="track-view-link"
                href="https://music.apple.com/us/artist/opeth/3196120?uo=4"
              >
                <p
                  class="T__StyledText-gjlic1-0 cbAvsY"
                  data-testid="view-track-text"
                >
                  View Track
                </p>
              </a>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
